vpr/src/route/rr_graph_storage.cpp

```diff
@@ -711,7 +711,13 @@ void t_rr_graph_storage::set_node_side(RRNodeId id, e_side new_side) {
     if (node_type(id) != IPIN && node_type(id) != OPIN) {
         VPR_FATAL_ERROR(VPR_ERROR_ROUTE, "Attempted to set RR node 'side' for non-channel type '%s'", node_type_string(id));
     }
-    node_storage_[id].dir_side_.side = new_side;
```

Comment:
I think this is not the right strategy.  I believe set_node_side should replace all existing sides (to match the description and previous behavior).  A new method, e.g. `set_node_sides` or a method like `add_node_side` should be added to handle the "more than one side case".

Reply:
Make senses to me. Will add the two new methods and rework the `set_node_side()`